third_party/move/move-vm/types/src/loaded_data/runtime_types.rs

```diff
@@ -624,25 +672,666 @@ impl fmt::Display for Type {
     }
 }
 
+/// Controls creation of runtime types, i.e., methods offered by this struct
+/// should be the only way to construct any type.
+#[derive(Clone)]
+pub enum TypeBuilder {
+    // Legacy configurations only limits type depth during type substitution. Will
+    // be removed in the nearest future when the new one stabilizes. The enum can
+    // be converted into the struct at that point.
+    Legacy,
+    New {
```

Comment:
Names like `New` are code smells :) It will stop being `New` at some point. Let's use a descriptive name?

Reply:
It will stop being new when we remove legacy (after the feature is enabled and this is deployed), and feature gating. Then it will be a struct with type size/depth, initialized based on gas parameters (which is versioned), and enum names will go away.

Happy to rename if you have suggestions. Maybe `WithLimits`?

Reply:
`WithLimits` is >>>> than `New` :)

Reply:
done